src/docselection.cpp

```diff
@@ -0,0 +1,169 @@
+        if (json.has<jsonxx::String>("measureRange")) {
+            m_measureRange = json.get<jsonxx::String>("measureRange");
+            if (m_measureRange == "all") {
+                m_selectionRangeStart = -1;
+                m_selectionRangeEnd = -1;
+            }
+            else if (m_measureRange.find("-") != std::string::npos) {
+                std::size_t pos = m_measureRange.find("-");
+                std::string startRange = m_measureRange.substr(0, pos);
+                std::string endRange = m_measureRange.substr(pos + 1, std::string::npos);
+                if (startRange == "start") {
+                    m_selectionRangeStart = -1;
+                }
+                else {
+                    startRange = std::regex_replace(startRange, std::regex(R"([^0-9])"), "");
+                    if (!startRange.empty()) m_selectionRangeStart = std::stoi(startRange);
+                }
+                if (endRange == "end") {
+                    m_selectionRangeEnd = -1;
+                }
+                else {
+                    endRange = std::regex_replace(endRange, std::regex(R"([^0-9])"), "");
+                    if (!endRange.empty()) m_selectionRangeEnd = std::stoi(endRange);
+                }
+            }
+            else {
+                std::string measureRangeInt = std::regex_replace(m_measureRange, std::regex(R"([^0-9])"), "");
+                if (!measureRangeInt.empty()) m_selectionRangeStart = std::stoi(measureRangeInt);
+                m_selectionRangeEnd = m_selectionRangeStart;
+            }
+            // Check values
+            if (m_selectionRangeStart == VRV_UNSET || m_selectionRangeEnd == VRV_UNSET
+                || (m_selectionRangeEnd != -1 && m_selectionRangeStart > m_selectionRangeEnd)) {
+                LogWarning("Selection 'measureRange' could not be parsed. No selection set.");
+                m_selectionRangeStart = VRV_UNSET;
+                m_selectionRangeEnd = VRV_UNSET;
+                return false;
+            }
+        }
+        else {
+            if (!json.has<jsonxx::String>("start") || !json.has<jsonxx::String>("end")) {
+                LogWarning("Selection requires 'start' and 'end'. No selection set.");
+                return false;
+            }
+            m_selectionStart = json.get<jsonxx::String>("start");
+            m_selectionEnd = json.get<jsonxx::String>("end");
+        }
+        return true;
+    }
+}
+
+void DocSelection::Set(Doc *doc)
+{
+    assert(doc);
+
+    m_isPending = false;
+
+    doc->m_selectionStart = "";
+    doc->m_selectionEnd = "";
+
+    if (m_selectionRangeStart != VRV_UNSET) {
+        std::string selectionStartId = "";
+        std::string selectionEndId = "";
+
+        ListOfConstObjects measures;
+        ClassIdComparison matchType(MEASURE);
+        // Depth is this->pages->page-system->measure
+        doc->FindAllDescendantsByComparison(&measures, &matchType, 4);
```

Comment:
Could also use `FindAllDescendantsByType(MEASURE, false)` here. Then you don't have to worry about search depth.